CMakeLists.txt

```diff
@@ -806,6 +806,35 @@ add_custom_target(check
   COMMAND ${CMAKE_CTEST_COMMAND}
   DEPENDS tests)
 
+# Catch2 C++ Unit Test support:
+option(WITH_CATCH2 "Enable Catch2 unit tests" ON)
+option(PREFER_SYSTEM_CATCH2 "Enable Catch2 unit tests (use system packages, if found)" OFF)
+option(WITH_SYSTEM_CATCH2 "Enable Catch2 unit tests (require system packages)" OFF)
+
+if(WITH_CATCH2 OR WITH_SYSTEM_CATCH2 OR PREFER_SYSTEM_CATCH2)
```

Comment:
to be consistent with other ceph dependencies, `WITH_CATCH2=OFF` should disable the dependency entirely

initially, we aren't sure that this won't break some developer builds, so people may want to disable it as a workaround. but once we start adding test coverage with catch2, we'd want those tests to always run. so i'd argue that, like gtest, it should not be optional in the long run

Reply:
1) ok, will fix! Thank you.
2) Agreed-- testing things where we have conditional switches like that is exponential growth and will miss, rather than... well... catch bugs!